tests/wallet/test_wallet_node.py

```diff
@@ -0,0 +1,297 @@
+import sys
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import pytest
+from blspy import PrivateKey
+
+from chia.util.config import load_config
+from chia.util.keychain import Keychain, generate_mnemonic
+from chia.wallet.wallet_node import WalletNode
+from tests.core.util.test_config import root_path_populated_with_config
+from tests.setup_nodes import test_constants
+
+
+# Can remove once root_path_populated_with_config is moved to conftests.py
+def stub() -> Any:
+    return root_path_populated_with_config
+
+
+@pytest.mark.asyncio
+async def test_get_private_key(root_path_populated_with_config: Path, get_temp_keyring: Keychain) -> None:
+    root_path: Path = root_path_populated_with_config
+    keychain: Keychain = get_temp_keyring
+    config: Dict[str, Any] = load_config(root_path, "config.yaml")["wallet"]
```

Comment:
I'll change this, but what's the rationale for using this other form when the function explicitly supports returning a `sub_config`? This is another case where we're moving an implementation detail to the caller instead of having it remain internal to the function (granted, it's just simple dictionary traversal in this case). A quick scan of the code shows that we mostly using the form in your suggestion, but that seems like an odd choice.

Reply:
I don't care much i just saw this, remembered we are usually using this parameter instead so i suggested it :)

Reply:
lol, I just realized that I was arguing the case _for_ the change you suggested :)

Reply:
Either is valid and I don't mean to try to sway the path, but my understanding is that we don't really have any concept of a subconfig other than that indirection of the indexing.  And I don't even think that indexing should be indexing at all, it should be attribute access on a class that has well-defined our config form.  `:]`  So my take is that using the parameter is just a confusing thing that misleads into thinking there's more to a 'subconfig' than there is.

But really, the parameter exists and is used elsewhere and I don't really care which form is used here.